src/helpers/bookmarkModel.js

```diff
@@ -0,0 +1,139 @@
+import Models from '../db/models';
+import ArticleModel from './articles';
+import errorResponse from '.';
+
+const { Article, Bookmark } = Models;
+
+/**
+ * @description class representing Bookmark Helpers
+ * @class BookmarkModel
+ */
+class BookmarkModel {
+  /**
+   * @description - This method is responsible for bookmarking an article
+   * @static
+   * @param {object} response - Response sent from the controller
+   * @param {object} slug - Request sent to the middleware
+   * @param {object} userId - Request sent to the middleware
+   * @returns {object} - object representing newly bookmarked article
+   * @memberof BookmarkModel
+   */
+  static async addBookmark(response, slug, userId) {
+    try {
+      const foundArticle = await ArticleModel.queryForArticle(slug);
+      const articleId = foundArticle.id;
+      await Bookmark.findOrCreate({
+        where: {
+          userId,
+          articleId
+        }
+      }).spread((check, created) => {
+        if (!created) {
+          return response.status(400).json(errorResponse(['Article already bookmarked']));
```

Comment:
Wait.

I thought the whole point of having a `BookmarkModel` helper was to use these helper methods to interface with the database then passing the data gotten from the db to the controllers which then do the actual `response.json()`s?

In a previous PR you raised, that was how it was implemented. Why is the reverse the case with this PR?

Reply:
Yeah, I know I did, but if you look closely, you would notice I was still passing the response as a parameter to the method. I just wanted to find a way for the controller to look as simple as possible. Thank you for spotting that out, I was able to do some refactoring.